starter-core/src/test/groovy/io/micronaut/starter/feature/function/awslambda/AwsLambdaSpec.groovy

```diff
@@ -495,22 +476,23 @@ tasks.named<io.micronaut.gradle.docker.NativeImageDockerfile>("dockerfileNative"
         )
         String build = output['build.gradle']
 
+        BuildTestVerifier verifier = BuildTestUtil.verifier(BuildTool.GRADLE, build)
+
         then:
         build.contains('runtime("lambda_provided")')
-        if (applicationType == ApplicationType.DEFAULT) {
-            assert !build.contains('implementation("io.micronaut.aws:micronaut-function-aws-custom-runtime")')
-        } else if (applicationType == ApplicationType.FUNCTION) {
-            assert build.contains('implementation("io.micronaut.aws:micronaut-function-aws-custom-runtime")')
-        }
-        !build.contains('implementation "io.micronaut:micronaut-http-server-netty"')
-        !build.contains('implementation "io.micronaut:micronaut-http-client"')
+        verifier.hasDependency("io.micronaut.aws", "micronaut-function-aws-custom-runtime", Scope.COMPILE) == (applicationType == ApplicationType.FUNCTION)
+        !verifier.hasDependency("io.micronaut", "micronaut-http-server-netty", Scope.COMPILE)
+
+        // Specifically added in io.micronaut.starter.feature.other.HttpClientTest.apply
+        verifier.hasDependency("io.micronaut", "micronaut-http-client")
```

Comment:
Previously we were trying to test that we _didn't_ have this dependency.

[But in we now apply it all the time if we have a custom runtime or graal](https://github.com/micronaut-projects/micronaut-starter/blob/d2b79da8d452557b656e808b9518e197fda92a5f/starter-core/src/main/java/io/micronaut/starter/feature/other/HttpClientTest.java#L75)

@sdelamo Is this ok?